File: Secre/Nueva carpeta/IdeaProjects/EmpresasTecnologicasMalaga-Angular-SpringBootJPA-master/src/main/java/org/iesbelen/empresastecnologicas/empresastecnologicasmalaga/domain/Empresa.java
package org.iesbelen.empresastecnologicas.empresastecnologicasmalaga.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name="empresa")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Empresa {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id_empresa")
    private long idEmpresa;
    private String nombre;
    private String direccion;
    private String telefono;
    private String email;
    private String web;
    @Column(length = 500)
    private String descripcion;
    private String imagen;
    private double latitud;
    private double longitud;
    @ManyToOne()
    @JoinColumn(name = "id_distrito", nullable = false, foreignKey = @ForeignKey(name = "FK_EMPRESA_DISTRITO"))
    private Distrito distrito;


}

File: Secre/Nueva carpeta/IdeaProjects/EmpresasTecnologicasMalaga-Angular-SpringBootJPA-master/src/main/frontend/app-empresasTecnologicas/src/app/models/empresa.ts
import {Distrito} from "./distrito";

export interface Empresa{
  idEmpresa: number;
  nombre: string;
  direccion: string;
  telefono: string;
  email: string;
  web: string;
  descripcion: string;
  imagen: string;
  latitud: number;
  longitud: number;
  distrito: Distrito;
}



